docs/data-structures/next-greatest-to-right.mdx

```diff
@@ -0,0 +1,134 @@
+---
+title: Next Greatest Element to the Right
+slug: /data-structures/next-greatest-to-right
+---
+
+## Introduction
+
+Stack as a data structure has many applications, especially because of its 'Last In, First Out' principle. One of them is to find the greatest element to the right of each element in the array. 
+
+## Algorithm
+
+- Initialize a stack. 
+- Start the scanning of the array from the right. 
+- Declare an array and a stack.
+- If the stack is empty, then no element to the right is greater than the present element
+- If the stack is not empty but the element present at the top of the stack is not greater than the current element then-
+    - Pop the stack till the stack is not empty or the top element of stack is greater than the current element.
+- If the stack is not empty and the element at the top of the stack is greater than the current element then
+    - push the topmost element to the stack.
```

Comment:
Capitalize the first word of a sentence always

Reply:
Done